feat: add react-native code to guide - OKTA-233817 (#1093)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/primary-authentication/react-native/primaryauth.md

### Build a sign in form

> To create a custom UI for user sign in, you need to first create a simple form to prompt the user for their username and password:

Reviewer: ```suggestion
To create a custom UI for user sign-in, you need to first create a simple form to prompt the user for their username and password:
```

```javascript
class LoginScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = { username: '', password: '' };
    this.login = this.login.bind(this);
  }

  login() {
    // TODO in Primary authentication section
  }

  render() {
    return (
      <SafeAreaView>
        <Text>Native Sign-In</Text>
        <View style={{ flexDirection: 'column' }}>
          <TextInput
            placeholder="User Name"
            onChangeText={username => this.setState({ username })}
          />
          <TextInput
            placeholder="Password"
            secureTextEntry={true}
            onChangeText={password => this.setState({ password })}
          />
          <View style={{ marginTop: 40, height: 40 }}>
            <Button
              testID="loginButton"
              onPress={this.login}
              title="Login"
            />
          </View>
        </View>
      </SafeAreaView>
    );
  }
}
```

### Primary authentication

> The SDK supports primary authentication flow since version 1.4.0. Implementing primary authentication can be as simple as calling `signIn` function from the SDK.

Reviewer: ```suggestion
The SDK supports the primary authentication flow since version 1.4.0. You can implement primary authentication by calling the `signIn` function.
```

> The code will continue work on the `login` method from last section.

Reviewer: I'm not sure what this sentence means.

> The following code example is a continuous work of the `login` method from the last section.

Reviewer 2: is a continuation of the `login` method from the last section.
